### DISCUSSION TITLE
How I Use Stacked Git at $WORK

### DISCUSSION THREAD

**Discussion by lthms** [stacked-git/stgit#177] [Show and tell]
Hi all.

I have been using Stacked Git in my daily workflow for a few months now, and wanted to share the workflow I’ve created for myself.

I’ve finally taken the time to [write an article about it](https://soap.coffee/~lthms/opinions/StackedGit.html)! and I’m curious to hear your thoughts on this.

Besides, this is for me the perfect opportunity to thank you all for your hard work. I really appreciate this software :heart:

**Comment by jpgrayson** [comment]
Hi Thomas. Great article. I appreciate both the advocacy for StGit and, as a maintainer, hearing more about how StGit is used in the wild. Thank you for taking the time and effort to write this!

Regarding the section "Stacked Git Feels Slow", I agree both in that StGit feels too slow for me too and that the root cause, at least for many StGit commands, is Python startup overhead. This is something I've been working on for the past several months. I'll have more on this development at a later date, but initial performance improvements are in the 2-4x range.

**Reply by lthms** [reply]
I can’t wait to try it out! Thanks again for the hard work. :)

**Reply by claxclinton** [reply]
Thanks for the hard work of maintaining StGIT! I'm using it heavily and know several others who use it.
Really looking forward to the speedup though.

**Reply by lthms** [reply]
Hi @jpgrayson. Any news on the speedup improvement work? If I can give a hand, eg, by testing, I’d love to!

**Reply by jpgrayson** [reply]
@lthms, @claxclinton , here is what I've been working on to speed up rust: https://github.com/stacked-git/stgit/discussions/185. I'd love for your feedback.

**Reply by lthms** [reply]
I have built it, and start using it. This made my prompt so much more enjoyable to us (compare to the latency before).

**Comment by claxclinton** [comment]
Great! Thanks Pete!
I’ve checked it out and announced it to my colleagues. We’ve got some some real stiff veterans at my workplace. I’m not fluid in rust unfortunately but others are.

I’ve checked it out and built it. Very fast!
When using from Emacs and showing a patch diff I saw an error:

The argument ‘—diff opts <OPTIONS>‘ was provided more than once.

Sorry, I’ll make a bug report out of it later.
I can’t get into my private account from works.

Thanks again, great work!

MVH Claes

**Comment by lthms** [comment]
Hi everyone.

One year to the day has passed since I have posted this blogpost, so I’ve figured: [why not post an update?](https://soap.coffee/~lthms/opinions/StackedGit2.html) :)

Happy new years, and thanks again for all the good work you are doing on Stacked Git!

**Reply by claxclinton** [reply]
Happy new year, and I also say big thanks for the great rewite!

**Reply by jpgrayson** [reply]
I also very much appreciate your writings about StGit. Your branchless approach is similar to how I've been using StGit for years, so it's nice to have a written-out description of it.

And thanks for the praise on StGit 2.0's speed improvements. As an FYI, StGit 2.2 will be even faster for the interrogation commands (e.g. `stg series`, `stg top`, etc.) typically used in command prompts due to switching from git2-rs to gitoxide.

Is it okay if I link your blog post from the [StGit homempage](https://stacked-git.github.io)?

**Reply by jpgrayson** [reply]
I should also note that the concept of patch dependencies has crossed various people's minds in the past and appears in StGit's [TODO file](https://github.com/stacked-git/stgit/blob/4245bdb591adbf1caf82434cb808b4b8648fe217/TODO#L1).

AFAIK, there is no design on the table for how this might work, so it remains a greenfield.

**Reply by lthms** [reply]
> Is it okay if I link your blog post from the [StGit homempage](https://stacked-git.github.io/)?

Be my guest! It would be an honor :)

> I should also note that the concept of patch dependencies has crossed various people's minds in the past and appears in StGit's [TODO file](https://github.com/stacked-git/stgit/blob/4245bdb591adbf1caf82434cb808b4b8648fe217/TODO#L1).

It would be very interesting to see a patch theory thingy implemented on top of StGit instead of as a brand new DVCS like Darcs or Pijul. You wouldn't get the same level of insurance, I have no doubt, but you would keep Git ecosystem.
